Test/prover/mocksolver.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+while true ; do
+    read cmd
+    case $cmd in
+      *get-info*name*   ) echo "(:name \"Z3\")" ;;
+      *get-info*rlimit* ) echo "(:rlimit bad)" ;;
```

Comment:
Cute way to reproduce the bug, nice :) 

I'm assuming the real-world behaviour is a random error message of some kind? Would it be smarter for the system to check for that earlier? I think making `GetRCount` more robust is smart anyway, but are we swallowing a potentially useful error message?

Reply:
Yes, I expect so. I haven't been able to reproduce the situation well enough to know what sort of response it's actually getting from the solver. It may be possible to detect it earlier once we know what it is. And your next comment gets at that. :)